client/js/components/NavigationDrawer.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <v-navigation-drawer
+    :value="showNav"
+    class="primary"
+    dark
+    app
+    data-cy="nav-drawer"
+    :permanent="showNav"
+    :mini-variant="isCollapsed"
+  >
+    <!-- Page Links -->
+    <v-list>
+      <v-list-item
+        v-for="({ text, icon, page }, i) in pageLinks"
+        :key="i"
+        link
+        exact
+        :to="page"
+        :data-nav="text"
+      >
+        <v-icon class="mr-4"> mdi-{{ icon }} </v-icon>
+        {{ text }}
+      </v-list-item>
+    </v-list>
+    <!-- Expand/Collapse -->
+    <template #append>
+      <v-list v-if="!isSmallDevice">
+        <v-list-item :data-cy="collapseMenuAttribute" @click="userHasCollapsed = !userHasCollapsed">
+          <v-icon class="mr-2">{{ collapseMenuIcon }}</v-icon>
+          <template v-if="!isCollapsed">Collapse Menu</template>
+        </v-list-item>
+      </v-list>
+    </template>
+  </v-navigation-drawer>
+</template>
+
+<script>
+export default {
+  name: 'NavigationDrawer',
+  data() {
+    return {
+      userHasCollapsed: false,
```

Comment:
I want to collapse the nav of the user hits the button or if we're on mobile (always collapsed on mobile without toggle option). 

So my intent here was to have the user control a Boolean in data and then control the nav drawer with a computed prop that collapses if the data or isMobile is true. Is that a sensible approach? And can you think of a way to make that more clear?

Reply:
I think that makes sense. In that case isCollapsed makes more sense, which is then driven by user action/mobile breakpoint.

Reply:
I have isCollapsed as the computed prop which is actually sent to the nav drawer, and isCollapsedByUser as the data the user controls. Reasonable? This actually seems like a case where composition api would be great so the data and computed prop could be next to each other instead of separated across option blocks